server/sonar-server/src/main/java/org/sonar/server/component/index/ComponentIndexQuery.java

```diff
@@ -22,25 +22,26 @@
 import java.util.Collection;
 import java.util.Collections;
 import java.util.Optional;
+import java.util.Set;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
 
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
 public class ComponentIndexQuery {
 
   private final String query;
-  private Collection<String> qualifiers = Collections.emptyList();
-  private Optional<Integer> limit = Optional.empty();
+  private final Collection<String> qualifiers;
+  private final Set<String> recentlyBrowsed;
+  @CheckForNull
+  private final Integer limit;
```

Comment:
`limit` can't be nullable. `int` type should be used with default value `SuggestionsAction#DEFAULT_LIMIT` (constant to be moved here IMO).